ppafm/common.py

```diff
@@ -17,56 +22,155 @@
-    "plotSliceTo": 22,
-    "plotSliceBy": 1,
-    "imageInterpolation": "bicubic",
-    "colorscale": "gray",
-    "colorscale_kpfm": "seismic",
-    "ddisp": 0.05,
-    "aMorse": -1.6,
-    "tip_base": np.array(["None", 0.00]),
-    "Rtip": 30.0,
-    "permit": 0.00552634959,
-    "vdWDampKind": 2,
-    "#": None,
-}
-# fmt: on
+class PpafmParameters(pydantic.BaseModel):
+    """Parameters for the PPAFM simulation.
+
+    The parameters description can be found on the wiki page: https://github.com/Probe-Particle/ppafm/wiki/Setting-simulation-parameters
+    """
+
+    PBC: bool = True
+    nPBC: typing.List[int] = [1, 1, 1]
+    gridN: typing.List[int] = [-1, -1, -1]
+    gridO: typing.List[float] = [0.0, 0.0, 0.0]
+    gridA: typing.List[float] = [20.0, 0.0, 0.0]
+    gridB: typing.List[float] = [0.0, 20.0, 0.0]
+    gridC: typing.List[float] = [0.0, 0.0, 20.0]
+    FFgrid0: typing.List[float] = [-1.0, -1.0, -1.0]
+    FFgridA: typing.List[float] = [-1.0, -1.0, -1.0]
+    FFgridB: typing.List[float] = [-1.0, -1.0, -1.0]
+    FFgridC: typing.List[float] = [-1.0, -1.0, -1.0]
+    moleculeShift: typing.List[float] = [0.0, 0.0, 0.0]
+    probeType: str = "O"
+    charge: float = 0.00
+    Apauli: float = 18.0
+    Bpauli: float = 1.0
+    ffModel: str = "LJ"
+    Rcore: float = 0.7
+    r0Probe: typing.List[float] = [0.00, 0.00, 4.00]
+    stiffness: typing.List[float] = [-1.0, -1.0, -1.0]
+    klat: float = 0.5
+    krad: float = 20.00
+    tip: typing.Union[str, typing.List] = "s"
+    sigma: float = 0.7
+    scanStep: typing.List[float] = [0.10, 0.10, 0.10]
+    scanMin: typing.List[float] = [0.0, 0.0, 5.0]
+    scanMax: typing.List[float] = [20.0, 20.0, 8.0]
+    scanTilt: typing.List[float] = [0.0, 0.0, -0.1]
+    tiltedScan: bool = False
+    kCantilever: float = 1800.0
+    f0Cantilever: float = 30300.0
+    Amplitude: float = 1.0
+    plotSliceFrom: int = 16
+    plotSliceTo: int = 22
+    plotSliceBy: int = 1
+    imageInterpolation: str = "bicubic"
+    colorscale: str = "gray"
+    colorscale_kpfm: str = "seismic"
+    ddisp: float = 0.05
+    aMorse: float = -1.6
+    tip_base: typing.List = ["None", 0.00]
+    Rtip: float = 30.0
+    permit: float = 0.00552634959
+    vdWDampKind: int = 2
+    Vbias: float = 0.0
+
+    class Config:
+        arbitrary_types_allowed = True
+        validate_assignment = True
+
+    @classmethod
+    def from_file(cls, file_path: typing.Union[str, Path] = Path("params.ini")):
+        """Load parameters from a file."""
+        file_path = Path(file_path)
+        object = cls()
+        suffix = file_path.suffix
+        try:
+            with open(file_path) as file:
+                if suffix == ".ini":
+                    lines = file.readlines()
+                    object.load_ini(lines)
+                elif suffix == ".yaml":
+                    object.apply_options(yaml.safe_load(file))
+                elif suffix == ".toml":
+                    object.apply_options(toml.load(file))
```

Comment:
Do we need to support both yaml and toml? Having too many options for doing the same thing might just create confusion. I would probably just pick one.

Reply:
We do not, it was just so easy to implement all of them.

I would pick `toml` if we want to support only one. It is very close to the original "ini" format. Also, starting from Python 3.11 is a part of the official Python distribution, so there is no need for an extra module.

Maybe let's hear @ProkopHapala 's opinion here?

Reply:
I have no experience with either, but being part of standard python distribution is strong argument for me. My big concern is to minimize external dependencies because I often angry when I install something ending up in dependency hell.

Reply:
It looks like the majority leans towards only supporting `toml` format. Then it is decided!

Reply:
done in https://github.com/Probe-Particle/ppafm/pull/275/commits/2ba6441df8253fba6b0d931734e0a74f6f311ab9